plugins/woocommerce/includes/admin/helper/class-wc-helper-admin.php

```diff
@@ -58,28 +58,28 @@ public static function add_marketplace_settings( $settings ) {
 	 * @return string
 	 */
 	public static function get_connection_url() {
+		global $current_screen;
+
+		$connect_url_args = array(
+			'page'                 => 'wc-addons',
+			'section'              => 'helper',
+		);
+
 		// No active connection.
-		if ( ! WC_Helper::is_site_connected() ) {
-			$connect_url = add_query_arg(
-				array(
-					'page'              => 'wc-addons',
-					'section'           => 'helper',
-					'wc-helper-connect' => 1,
-					'wc-helper-nonce'   => wp_create_nonce( 'connect' ),
-				),
-				admin_url( 'admin.php' )
-			);
-
-			return $connect_url;
+		if ( WC_Helper::is_site_connected() ) {
+			$connect_url_args['wc-helper-disconnect'] = 1;
+			$connect_url_args['wc-helper-nonce']      = wp_create_nonce( 'disconnect' );
+		} else {
+			$connect_url_args['wc-helper-connect'] = 1;
+			$connect_url_args['wc-helper-nonce']   = wp_create_nonce( 'connect' );
+		}
+
+		if ( ! empty( $current_screen->id ) && 'woocommerce_page_wc-admin' === $current_screen->id ) {
```

Comment:
I personally like replacing `! empty` with `isset`. I find it's easier since it doesn't use negation

```suggestion
		if ( isset( $current_screen->id ) && 'woocommerce_page_wc-admin' === $current_screen->id ) {
```